src/main/java/bio/terra/workspace/service/datareference/utils/DataReferenceValidationUtils.java

```diff
@@ -68,4 +84,42 @@ private void validateSnapshotReference(SnapshotReference ref, AuthenticatedUserR
               + " Verify that your reference was correctly defined and the instance is correct");
     }
   }
+
+  /**
+   * Validates that a bucket exists and the user has GET access to it. This makes no other
+   * assumptions about the bucket.
+   */
+  private void validateGoogleBucket(GoogleBucketReference ref, AuthenticatedUserRequest userReq) {
+    try {
+      // StorageCow.get() returns null if the bucket does not exist or a user does not have access,
+      // which fails validation.
+      Optional.ofNullable(crlService.createStorageCow(userReq).get(ref.bucketName()))
+          .orElseThrow(
+              () ->
+                  new InvalidDataReferenceException(
+                      "Could not access specified GCS bucket. Ensure the name is correct and that you have access."));
```

Comment:
include the bucket name in these messages.

Reply:
same logging/response issue as above